### PR TITLE
Add Splunk HEC exporter example

### PR DESCRIPTION
**Pull request by atoulme**
**Description:** 
Add a Splunk HEC exporter example

**Testing:**
N/A

**Documentation:**
Example comes with a README, reproduced below:

# OpenTelemetry Collector Splunk demo

This demonstration shows how the OpenTelemetry Collector can send data to a Splunk instance.

Splunk runs as part of this docker-compose demo to show metrics data from your machine. .

To build and run the demo, switch to this directory and run

`docker-compose up`

Hit `http://localhost:18000` and log in with `admin`/`opentelemetry`, then go to the [analytics workspace](http://localhost:18000/en-US/app/search/analytics_workspace).

You will see a list of analytics exposed by the host metrics collector, which captures simple metrics exposed inside the OpenTelemetry Collector Docker container.

metrics

### CODE REVIEW

**Review by tigrannajaryan — COMMENTED** [review]
LGTM, but would like @bogdandrutu to also have a look before merging.

**Review by bogdandrutu — COMMENTED** [review]
I would suggest to have this in "exporter/splunk/example" and try to avoid pictures of the product if possibe.

File: .circleci/config.yml [hunk 1]
```diff
@@ -106,6 +106,12 @@ workflows:
           filters:
             tags:
               only: /.*/
+      - build-examples-splunkhecexporter:
```
Comment (tigrannajaryan) [inline .circleci/config.yml:109]: Can we avoid having component specific entries in CircleCI? It is best to add this to main makefile as a target that iterates over components and executes corresponding targets in component makefiles if they exists.

Comment (atoulme) [reply .circleci/config.yml:109]: Tigran, you assume I am a competent developer. While I believe I understood your comment, I might not be best placed to perform this change. I'll try though.

Comment (atoulme) [reply .circleci/config.yml:111]: I have changed the makefile - just not sure how to iterate over components.

Comment (tigrannajaryan) [reply .circleci/config.yml]: This is good enough, we can worry about it when we have more examples.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**